src/test/java/com/github/fridujo/rabbitmq/mock/ChannelTest.java

```diff
@@ -633,4 +637,126 @@ void commit_or_rollback_can_be_called_multiple_times_after_a_single_select() thr
+                                               Envelope envelope,
+                                               AMQP.BasicProperties properties,
+                                               byte[] body) {
+                        assertThat(reply.compareAndSet(null, new String(body))).isTrue();
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void handleCancelOk(String consumerTag) {
+                        cancelled.set(true);
+                    }
+                });
+
+                latch.await(1, TimeUnit.SECONDS);
+                channel.basicCancel(consumerTag);
+
+                assertThat(cancelled).isTrue();
+                assertThat(reply.get()).isEqualTo("pong");
+            }
+        }
+
+        // assert that internal queue is removed
+        try (Connection conn = new MockConnectionFactory().newConnection()) {
+            try (Channel channel = conn.createChannel()) {
+                assertThrows(IOException.class, () -> {
+                    channel.queueDeclarePassive(replyTo);
+                });
+            }
+        }
+        
+    }
+
+    @Test
+    void directReplyTo_basicConsume_noAutoAck() {
+        assertThrows(IllegalStateException.class, () -> {
+            try (Connection conn = new MockConnectionFactory().newConnection()) {
+                try (Channel channel = conn.createChannel()) {
+                    channel.basicConsume("amq.rabbitmq.reply-to", false, new DefaultConsumer(channel) {
+                        @Override
+                        public void handleDelivery(String consumerTag,
+                                                   Envelope envelope,
+                                                   AMQP.BasicProperties properties,
+                                                   byte[] body) {
+
+                            fail();
+                        }
+
+                        @Override
+                        public void handleCancelOk(String consumerTag) {
+                            fail();
+                        }
+                    });
+                }
+            }
+        });
+    }
+
+    @Test
+    void directReplyTo_basicGet_noAutoAck() {
+        assertThrows(IllegalStateException.class, () -> {
+            try (Connection conn = new MockConnectionFactory().newConnection()) {
+                try (Channel channel = conn.createChannel()) {
+                    channel.basicGet("amq.rabbitmq.reply-to", false);
+                }
+            }
+        });
+    }
+    
+    @Test
+    void directReplyTo_replyTo_queue_exists() throws IOException, TimeoutException {
+        try (Connection conn = new MockConnectionFactory().newConnection()) {
+            try (Channel channel = conn.createChannel()) {
+                assertThat(channel.queueDeclarePassive("amq.rabbitmq.reply-to")).isNotNull();
+            }
+        };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/app/ledoyen/rabbitmq-mock/pullRequest?prid=4732226)